Added support for Solus, Old Fedora-based OS' + Interactive approach

AktasC opened the pull request:
Hi,

Hope you will enjoy this version more than the previous one.

Regards,
Cebrail Aktas, aktas_c

kayofeld commented on installation/install:
```diff
@@ -85,11 +85,13 @@ function handle_error
 	which pacman &> /dev/null && os="archlinux"
-	which dnf &> /dev/null && os="fedora"
 	which apt-get &> /dev/null && os="debian"
+	which apt-get &> /dev/null && os="deepin"
```
deepin seems to be a debian based system, it should be handled as such, and no line should be added for it
`which apt-get &> /dev/null && os="debian"
+	which apt-get &> /dev/null && os="deepin"`
if you have an Ubuntu, Mint, Debian (etc..) it will choose deepin, but they are debian based systems ;)

AktasC replied:
Yes Deepin is debian based but I didn't know it would have been handled automatically, my bad :)
I will try to add more useful support 👍

kayofeld commented on installation/install:
```diff
@@ -115,21 +117,27 @@ function sys_upgrade
 		debian)
 			sudo apt-get update; sudo apt-get upgrade
 			;;
+		deepin)
```
Same here as above

AktasC replied:
Gonna upload that fix right now 👍

AktasC commented:
At first I didn't even care about that kind of scripts, now I wanna take part in it ! 💯
Will give a try to an interactive version of this script and let you know when it's done ! 😃

bew commented:
Cool, thanks @AktasC I'll be happy to review if you do something!

bew commented on installation/install:
```diff
@@ -141,21 +149,27 @@ function sys_install
 				;;
+			deepin)
+				echo "apt-get install"
+				;;
```
Same here, you can remove this block

AktasC commented:
Check my latest push, tell me what you think of it 👍

bew commented:
Well.. You destroyed indentation in most part of the code ^^ be careful when you commit, you should review what will be commited before doing the actual commit (with git diff)... 

For the interactive part, please open an other issue and make a new pull request! 

This PR is for adding support to solus only! Don't mix everything please :)

AktasC commented:
First of all :
Sorry for the indentation. I'm using Atome and for me it looks "normal".
I added interactive choice between zsh and fish.

That being said, I also added support for older Fedora based OS which use YUM instead of DNF.
For these, I also added the --security flag so that this script doesn't cause any problem with their stability.

Actually sorry for mixing things up, I'm used to work alone on my projects. I rarely take part in/or modify other people's project.

What shall I do ? 😱 

PS : For now I changed the name so it fits the topic 😆

bew commented:
I don't know what @kayofeld think about that, but in my opinion we should split your work in multiple PR and review + merge them one by one. 
I'll wait for his point of view before doing anything

I'm dropping this here for you to read as a starting point:
http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/

AktasC commented:
Hi I'm back,

I've been so much in a good mood that I even included VIM/Emacs interactive selection + Epitech configuration but didn't wanna push it so it doesn't break the current PR..
I think for now we should close this one and wait for @kayofeld to whether accept or decline my future PR which will include several more features..

Best regards,
Cebrail Aktas, Aktas_C

kayofeld commented:
Every improvement is good, but you will need to open new branches for each module added, it is then easier to test it.
Plus it would be good to work on another branch than master. Since it is the release branch, I want to test everything before it gets merged into it. (@Bew78LesellB being also able to do it.)
Concerning zsh vs fish vs bash, instead of choosing for the user, why not give the user a choice ? :)

Take a look at https://github.com/kayofeld/script-installation-ordinateur-epitech/issues/4 , I will take the time to work on it on march, but if you are motivated, why don't you do it ? :)

Thanks

AktasC commented:
@kayofeld Coucou ! :D
Désolé les gars mais j'ai mangé 12h d'anglais aujourd'hui je suis en overdose.

Donc je vais reprendre ici point par point les différentes choses que j'ai mis en place histoire que ce soit plus clair à comprendre pour nous trois un tant soit peu.

Dans un premier temps, le système de branches + issue + pull request me pose trop de contraintes du coup j'ai préféré fork puis bosser de mon côté à ma vitesse..

Dans un second temps, j'ai ajouté l'interactivité tant demandée au niveau de l'éditeur, du terminal et du shell.
Grosso modo on a le choix entre emacs/vim, fish/zsh et terminator/rxvt-unicode. Cela ne veut pas dire que l'on force l'utilisateur à les utiliser bien entendu, Gnome-Terminal est installé sur toutes les distros utilisant un WM basé sur Gnome.
J'en ai profité pour inclure les fichiers de configuration de cyril_l pour vim.

Troisièmement, petite remarque personnelle : 
Qu'est-ce que j'ai pu faire de la merde .. Certainement une accumulation de fatigue mais quand je vois le mélange entre sh et C que j'ai fait je me dis que je mérite un renvois définitif de Github x)

Let's finish that long comment in english : 
Thanks for all the contribution and inspiration, I honestly wouldn't have lifted up my lazy ass to start all that on my own.

Best regards,
Cebrail.
